### PR TITLE
.NET Standard support

### PR DESCRIPTION
**Pull request by gentledepp**
Hi!
We are using Cake 0.22.0 in our build, which requires net46.
As Cake.AndroidAppManifest runs on net452, I upgraded it.

I also upgraded to the latest csproj format along with the Nuget 4.0 "<PackageReference />" style.

In the cake script, I changed the NuGet package creation from using "NuGetPack" to "MSBuild", because 
* the metadata is in the csproj anyways
* and it automagically adds the correct NuGet package dependencies

One last thing: I did not manage to get "GitLink" to run. Always got the exception

> Running GitLink for ./src/Cake.AndroidAppManifest.sln
> error: option -u requires a value
> 
... which did not make sense to me.
So would you mind checking out if I did not destroy something, and (hopefully) also explain to me what I did wrong?

Cheers

### CODE REVIEW

**Comment by jzeferino** [comment]
@gentledepp I will take a look at this in the weekend. Thank you.
The GitLink issue was already reported by me [here](https://github.com/GitTools/GitLink/issues/176), and i think I will remove it until its solved.

File: src/Cake.AndroidAppManifest_.nuspec [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (jzeferino) [inline src/Cake.AndroidAppManifest_.nuspec:1]: @gentledepp since the nuget package info is in the .csproj, any reason why you added this .nuspec here?

**Comment by gentledepp** [comment]
I don't think I did that deliberately. Guess it can go. Sorry for that

Am 21.10.2017 11:31 schrieb "jzeferino" <notifications@github.com>:

**Comment by jzeferino** [comment]
@gentledepp I can't build the Test project:

> /Users/jz/Desktop/Cake.AndroidAppManifest/src/Cake.AndroidAppManifest.Tests/CSC: Error CS0006: Metadata file '/Users/jz/Desktop/Cake.AndroidAppManifest/src/Cake.AndroidAppManifest/bin/Debug/net46/Cake.AndroidAppManifest.dll' could not be found (CS0006) (Cake.AndroidAppManifest.Tests)

The Cake.AppManifest its not being generated into net46 folder.

**Comment by jzeferino** [comment]
@gentledepp since we have some users that need this ASAP, I will only update the .NET Framework from 452 to  46 and after that we could move to the .netstandard. Seems cool for you?

**Comment by gentledepp** [comment]
yeah that's fine with me.
Sorry - v'been quite busy this weekend. I may have a chance to look at this issue on wednesday, though.

**Comment by jzeferino** [comment]
@gentledepp I've made some big changes now, if you don't mind please do a new PR so we can work together converting the project to .netstandard 1.6 and add better packaging.

**Comment by gentledepp** [comment]
Hi! I do not quite understand the problem with .net46 you're facing.
On my machine everything builds just fine, be it with VS2017 or with cake.

I also added the recommended build.ps1 file of cake, so you can now run specific targets using 
> build.ps1 <TheTargetName>
e.g.
> build.ps1 Package

as for the .net46 problem: Do you have the required sdks installed?

**Comment by gentledepp** [comment]
So, I now merged with the master branch.
Please run the build like so:
> .\build.ps1 package
.. to see if it actually works.
If you have problems building this solution with VS, please make sure you have the latest and greatest version of VS2017! (not because of .NET Standard, but because of the new csproj format)
**Note:** As NSubstitute caused me a lot of headaches as it caused all test-runs in cake to fail, and as it was not really used anyway, I removed it from the equation.

**Comment by jzeferino** [comment]
@gentledepp thanks for the effort. We have a problem now, Since i've updated and released the addin with .net46, as I said #9. This PR should is behind that changes.
You could rebase develop, or re-done the PR.

At this moments this is a bit confusing. Do you mind to start a new feature from develop and apply the .netstandar16 changes on top of it and when your done PR it to develop again?

**Comment by jzeferino** [comment]
@gentledepp I will take some of your changes and create a new PR with .NET Standard support, cool for you?

**Comment by gentledepp** [comment]
Hi!
Sorry - I am quite busy at the moment.
I could find some time on thursday.
Then I would cancel my master PR, pull from your dev branch and make a new PR on dev
would that be ok with you?

**Comment by jzeferino** [comment]
@gentledepp ok, I will wait for it and after you made it I will help.

**Comment by jzeferino** [comment]
@gentledepp any news?

**Comment by gentledepp** [comment]
Just created another PR targeting the develop branch.
SHould work fine.

**Comment by gep13** [comment]
@gentledepp just for future reference, it is possible to re-target a PR against another branch.  There is no hard requirement to create a new one 😄

**Comment by jzeferino** [comment]
yes @gep13 but this PR was a bit older since i've made some major changes in develop, he could rebase develop into de PR but, it will be more clean to start the PR again since it has some small changes. thanks.

**Pull request closed without merging**